packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -1203,6 +1208,11 @@ function LineplotViz(props: VisualizationProps<Options>) {
     setTruncatedDependentAxisWarning,
   ]);
 
+  // change default vizConfig.useBinning to be true if timeline Viz
```

Comment:
I think there are two ways to avoid using the effect (which I presume could cause extra renders)

1. figure out how to pass the `options` to the `createDefaultConfig` function (I have no idea how!)
2. instead of updating the `vizConfig.useBinning` state, just use a regular "variable".

```
  const useBinning = alwaysEnableUseBinning || vizConfig.useBinning;
``` 

Then replace all `vizConfig.useBinning` with `useBinning`

Reply:
But maybe it doesn't work if we are passing `vizConfig` whole to other functions...

Reply:
Like `getRequestParams` - bummer 😞

Reply:
@bobular For your #1, the problem is that createDefaultConfig is defined out of Lineplot Viz component, thus options cannot be used as it is passed to Lineplot Viz component. As for #2, yes that was the reason why I didn't use a local variable like const useBinning, as it could have caused other side effects.

Reply:
Can't we do something like this (completely untested):

```
  const customDefaultConfig = () => {
    const config = createDefaultConfig();
    if (options?.showMarginalHistogram) {
      return { ...config, useBinning: false };
    } else {
      return config;
    }
  };
  const [vizConfig, updateVizConfig] = useVizConfig(
    visualization.descriptor.configuration,
    LineplotConfig,
    customDefaultConfig,
    updateConfiguration
  );
```

You're correct that `createDefaultConfig` is exposed to the "outside" but I think it's only called in one place: (VisualizationsContainer.tsx:L377) and I don't think it's needed there (I will do a quick test).

Reply:
@bobular yes actually I have been doing similar thing to test as well 👍

Reply:
Yes it works fine with `configuration: undefined` at L377.

The visualization component fills in the default config if the stored configuration isn't "correct".

Reply:
Cool. But with `useBinning: true` of course!